eng/liveBuilds.targets

```diff
@@ -5,12 +5,29 @@
-    <LibrariesSharedFrameworkRefArtifactsPath Condition="'$(LibrariesSharedFrameworkRefArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'ref', 'microsoft.netcore.app', '$(LibrariesConfiguration)'))</LibrariesSharedFrameworkRefArtifactsPath>
-    <LibrariesAllRefArtifactsPath Condition="'$(LibrariesAllRefArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'ref', '$(NetCoreAppCurrent)'))</LibrariesAllRefArtifactsPath>
-    <LibrariesSharedFrameworkBinArtifactsPath Condition="'$(LibrariesSharedFrameworkBinArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'pkg', '$(NetCoreAppCurrent)', 'runtime', '$(LibrariesOSGroup)-$(LibrariesConfiguration)-$(TargetArchitecture)'))</LibrariesSharedFrameworkBinArtifactsPath>
-    <LibrariesAllBinArtifactsPath Condition="'$(LibrariesAllBinArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'runtime', '$(NetCoreAppCurrent)-$(LibrariesOSGroup)-$(LibrariesConfiguration)-$(TargetArchitecture)'))</LibrariesAllBinArtifactsPath>
-    <LibrariesNativeArtifactsPath Condition="'$(LibrariesNativeArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'native', '$(NetCoreAppCurrent)-$(LibrariesOSGroup)-$(LibrariesConfiguration)-$(TargetArchitecture)'))</LibrariesNativeArtifactsPath>
+
+    <CoreCLRArtifactsPath Condition="'$(CoreCLROverridePath)' != ''">$([MSBuild]::NormalizeDirectory('$(CoreCLROverridePath)'))</CoreCLRArtifactsPath>
+    <CoreCLRArtifactsPath Condition="'$(CoreCLRArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(LibrariesArtifactsPath)', 'bin', 'coreclr', '$(CoreCLROSGroup).$(TargetArchitecture).$(CoreCLRConfiguration)'))</CoreCLRArtifactsPath>
+
+    <CoreCLRSharedFrameworkDir>$([MSBuild]::NormalizeDirectory('$(CoreCLRArtifactsPath)', 'sharedFramework'))</CoreCLRSharedFrameworkDir>
+
+    <LibrariesArtifactsPath Condition="'$(LibrariesArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts'))</LibrariesArtifactsPath>
+
+    <LibrariesPackagesDir>$([MSBuild]::NormalizeDirectory('$(LibrariesArtifactsPath)', 'packages', '$(LibrariesConfiguration)'))</LibrariesPackagesDir>
+    <LibrariesShippingPackagesDir>$([MSBuild]::NormalizeDirectory('$(LibrariesPackagesDir)', 'Shipping'))</LibrariesShippingPackagesDir>
+    <LibrariesNonShippingPackagesDir>$([MSBuild]::NormalizeDirectory('$(LibrariesPackagesDir)', 'NonShipping'))</LibrariesNonShippingPackagesDir>
+
+    <LibrariesAllConfigurationsArtifactsPath Condition="'$(LibrariesAllConfigurationsArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts'))</LibrariesAllConfigurationsArtifactsPath>
```

Comment:
Why is this needed? These artifacts will end-up as packages in the `Shipping`/`NonShipping`, so I think we could just use `LibrariesShipping/NonShippingPackagesDir` properties.

Reply:
> These artifacts will end-up as packages in the `Shipping`/`NonShipping`

This happens for the platform-specific artifacts, but the AllConfigurations artifacts have to be downloaded to a transport directory due to overlaps with the platform-specific artifacts. (Specifically, it caused a build failure with the `docs` directory when I tried to put them both in the same place.)

This property is the entry point for the build to find the transport directory:

```yaml
  # Download allconfigurations bits to a separate location. It overlaps with the live libraries
  # download and causes conflict when unzipping to the same location.
  - ${{ if eq(parameters.useOfficialAllConfigurations, true) }}:
    - name: LibrariesDownloadPathAllConfigurations
      value: 'artifacts/transport/librariesallconfigurations'
    - name: LibrariesAllConfigurationsArtifactsPathArg
      value: /p:LibrariesAllConfigurationsArtifactsPath=${{ parameters.buildCommandSourcesDirectory }}$(LibrariesDownloadPathAllConfigurations)
```

The default is so things will (in theory) work if someone runs both the allconfigurations and platform-specific Libraries builds locally.

Reply:
I see what you're saying. Since we publish, ref, testhost and other stuff they end-up clashing. 

For official builds we could get away by just publishing /packages/ as part of the artifacts directory and downloading it into `artifacts` in the sources directory of the installer step, however, for local development there is indeed a clash.

What do you think? You think is better having this overhead of more properties that at the end, end-up pointing to the same location by default?

Reply:
> For official builds we could get away by just publishing /packages/ as part of the artifacts directory and downloading it into `artifacts` in the sources directory of the installer step,

We can still get clashes that only repro in the official build if, say, the allconfigurations and win-x64 legs start producing a nupkg with the same name. From a defensive perspective I like having the artifacts under completely separate roots for safety and easy traceability in the logs.

> You think is better having this overhead of more properties that at the end, end-up pointing to the same location by default?

IMO the extra properties are not significant vs. other considerations. I think the reason to go for "merging" this is to get local builds as close as possible to AzDO builds. (CoreCLR artifacts should be merged in as well to accomplish this!) But I think this is something that we can do incrementally, and I think it could really use some sort of backstop to make sure artifacts aren't stomping each other without us knowing about it.

Filed https://github.com/dotnet/runtime/issues/1030 for this.

Reply:
> We can still get clashes that only repro in the official build if, say, the allconfigurations and win-x64 legs start producing a nupkg with the same name. From a defensive perspective I like having the artifacts under completely separate roots for safety and easy traceability in the logs.

I see that very unlikely, and actually the rid specific legs, only produce runtime packages. All configurations when in an official build only produces OOB packages which are not build as part of a rid specific leg. We have been using the same drop for packages for over a year now for libraries builds in (corefx) because that's how we transported all the packages through artifacts to a single job that signed all the nupkgs.

> Filed #1030 for this.

Thanks for filing the issue, I agree that we should get local builds as close as possible to AzDO and have less overhead of passing global properties through command line arguments as that makes things harder to understand and get right.

Reply:
> I see that very unlikely, and actually the rid specific legs, only produce runtime packages.

Interesting, yeah, not too familiar with the background here when it comes to the corefx official build. In that case, the backstop is simply the official build failing in that very unlikely case, which is fine. I've added the AllConfigurations part in particular to the https://github.com/dotnet/runtime/issues/749 checklist, no need to flesh out https://github.com/dotnet/runtime/issues/1030 before doing that.

Reply:
Sounds good.